src/Platforms/AbstractPlatform.php

```diff
@@ -1533,6 +1564,14 @@ public function getCheckDeclarationSQL(array $definition): string
         $constraints = [];
         foreach ($definition as $def) {
             if (is_string($def)) {
+                Deprecation::trigger(
+                    'doctrine/dbal',
+                    'https://github.com/doctrine/dbal/pull/6805',
+                    'Passing column definition to %s() as string is deprecated. Pass the definition as array'
+                        . ' instead.',
+                    __METHOD__,
+                );
```

Comment:
None of the library code passes strings to `getCheckDeclarationSQL()`. If the code relies on this behavior, it can be implemented in the userland.

Likely, nobody uses those checks at all, but I don't want to deal with deprecating the whole thing right now. Checks are not populated as part of introspection, which would be challenging because they are arbitrary SQL expressions and cannot be easily parsed into the `min`/`max` column properties.